modules/evasion/windows/applocker_evasion_presentationhost.rb

```diff
@@ -0,0 +1,167 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Evasion
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Applocker Evasion - Windows Presentation Foundation Host',
+      'Description' => %(
+         This module will assist you in evading Microsoft
+         Windows Applocker and Software Restriction Policies.
+         This technique utilises the Microsoft signed binary
+         PresentationHost.exe to execute user supplied code.
+                        ),
+      'Author'      =>
+        [
+          'Nick Tyrer <@NickTyrer>', # module development
+          'Casey Smith' # presentationhost bypass research
+        ],
+      'License'     => 'MSF_LICENSE',
+      'Platform'    => 'win',
+      'Arch'        => [ARCH_X86],
```

Comment:
Does this not work on x64 for some reason? It doesn't seem like it would be an architecture-dependent module.

Reply:
I could not get x64 payloads to execute, this seems to be common issue as seen here: [https://social.msdn.microsoft.com/Forums/vstudio/en-US/695ca84a-73a2-468c-b125-da82834fd4d3/net-45-xbap-wpf-browser-application-on-windows-8-64-bit-desktop-mode?forum=wpf](url)